solver/utils.go

```diff
@@ -14,6 +19,48 @@ var (
 	differ = gojsondiff.New()
 )
 
+func getDocumentDiff(a, b *state.Document) (string, error) {
+	aCopy := a.ShallowCopy()
+	bCopy := a.ShallowCopy()
+	aContent := *a.Content
+	bContent := *b.Content
+	aCopy.Content = nil
+	bCopy.Content = nil
+	objDiff, err := getDiff(aCopy, bCopy)
+	if err != nil {
+		return "", err
+	}
+	var contentDiff string
+	if json.Valid([]byte(aContent)) && json.Valid([]byte(bContent)) {
+		aContent, err = pprintJSONString(aContent)
+		if err != nil {
+			return "", err
+		}
+		bContent, err = pprintJSONString(bContent)
+		if err != nil {
+			return "", err
+		}
+	}
+	edits := myers.ComputeEdits(span.URIFromPath("old"), aContent, bContent)
+	contentDiff = fmt.Sprint(gotextdiff.ToUnified("old", "new", aContent, edits))
+
+	return objDiff + contentDiff, nil
+}
+
+func pprintJSONString(JSONString string) (string, error) {
```

Comment:
Better to name functions what they do.
```suggestion
func prettyPrintJSONString(JSONString string) (string, error) {
```